service/history/shardContext.go

```diff
@@ -60,6 +60,8 @@ type (
 		UpdateTimerAckLevel(ackLevel time.Time) error
 		GetTimerClusterAckLevel(cluster string) time.Time
 		UpdateTimerClusterAckLevel(cluster string, ackLevel time.Time) error
+		GetLastShardSyncTimestamp() time.Time
```

Comment:
I think we should simplify the logic and have generation of sync task limited to replicator processor.  This way we don't need all these API changes to ShardContext.